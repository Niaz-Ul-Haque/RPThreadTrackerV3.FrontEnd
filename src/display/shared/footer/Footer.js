import React from 'react';

const Footer = () => {
	const currentYear = (new Date()).getFullYear();
	return (
		<footer className="app-footer">
			<a href="http://www.rpthreadtracker.com">RPThreadTracker</a> &copy; {currentYear} <a href="http://blackjack-software.com">Blackjack Software</a>
			<span className="float-right">Powered by <a href="http://coreui.io">CoreUI</a></span>
		</footer>
	);
};

export default Footer;
