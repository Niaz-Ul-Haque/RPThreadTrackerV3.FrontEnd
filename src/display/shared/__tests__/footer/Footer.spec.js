// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../../footer/Footer';
// #endregion imports

const DATE_TO_USE = new Date('June 5, 1989 03:24:00');
const MockDate = Date;
global.Date = jest.fn(() => DATE_TO_USE);
global.Date.UTC = MockDate.UTC;
global.Date.parse = MockDate.parse;
global.Date.now = MockDate.now;

describe('rendering', () => {
	it('should render valid snapshot', () => {
		const jsx = (<Footer />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
});
