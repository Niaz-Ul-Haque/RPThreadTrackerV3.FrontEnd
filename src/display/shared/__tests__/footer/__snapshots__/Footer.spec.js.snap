// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render valid snapshot 1`] = `
<footer
  className="app-footer"
>
  <a
    href="http://www.rpthreadtracker.com"
  >
    RPThreadTracker
  </a>
   Â© 
  1989
   
  <a
    href="http://blackjack-software.com"
  >
    Blackjack Software
  </a>
  <span
    className="float-right"
  >
    Powered by 
    <a
      href="http://coreui.io"
    >
      CoreUI
    </a>
  </span>
</footer>
`;
